jcg_evaluation/src/main/scala/JSFingerprintExtractor.scala

```diff
@@ -1,97 +1,91 @@
-              .map(f => new ExpectedCG(f))
-              .sorted(Ordering.by((f: ExpectedCG) => f.filePath))
-
-        if (config.debug) {
-            println("[DEBUG] expectedCGs:" + expectedCGs.map(_.filePath).mkString(","))
-            val generatedCGFiles = FileOperations.listFilesRecursively(adapterOutputDir, ".json").filter(f => f.getAbsolutePath.contains("js"))
-            println("[DEBUG] generatedCGFiles: " + generatedCGFiles.mkString(","))
-        }
-
-        val outputWriter = new BufferedWriter(getOutputTarget(outputDir))
+        outputDir.mkdirs()
+        val testDirs = new File(inputDir.getAbsolutePath).list().sorted
 
-        printHeader(outputWriter, expectedCGs.map(_.jsonFile))
+        val ow = new BufferedWriter(getOutputTarget(outputDir))
+        printHeader(ow, testDirs.map(new File(_)))
 
         for {
-            adapter <- adapters
-            algo <- adapter.possibleAlgorithms.filter(_.startsWith(config.algorithmFilter))
+            adapter <- config.adapters
+            cgAlgorithm <- adapter.possibleAlgorithms.filter(_.startsWith(config.algorithmFilter))
         } {
-            outputWriter.write(s"${adapter.frameworkName}-$algo")
-
-            println(s"Creating fingerprints for ${adapter.frameworkName} $algo")
-            val fingerprintFile = getFingerprintFile(adapter.frameworkName, algo, outputDir)
-            if (fingerprintFile.exists()) {
-                fingerprintFile.delete()
-            }
-            val fingerprintWriter = new PrintWriter(fingerprintFile)
+            ow.write(s"${adapter.frameworkName}-$cgAlgorithm")
 
-            for (expectedCG <- expectedCGs) {
-                val testName: String = expectedCG.filePath.split("/").last
-                if (config.debug) println("[DEBUG] Test name: " + testName + " " + algo)
-                val generatedCGFile = new File(adapterOutputDir, s"${adapter.frameworkName}/$algo/").listFiles().find(_.getName == testName).orNull
+            println(s"creating fingerprint for ${adapter.frameworkName} $cgAlgorithm")
+            val fingerprintWriter: PrintWriter = makeFingerprintWriter(outputDir, adapter, cgAlgorithm)
 
-                // if callgraph file does not exist write Error to result
-                var assessment: Assessment = Error
-                if (generatedCGFile != null) {
-                    val generatedCG = new AdapterCG(generatedCGFile)
-                    // check if call graph has missing edges
-                    val isSound = generatedCG.compareLinks(expectedCG).length == 0
-                    assessment = if (isSound) Sound else Unsound
+            // create output dir for every adapter-cgAlgorithm combination
+            val adapterDir = new File(s"$outputDir/${adapter.frameworkName}/$cgAlgorithm")
+            adapterDir.mkdirs()
+            for (testDir <- testDirs.filter(_.startsWith(config.projectFilter))) {
+                val future = Future {
```

Comment:
Most of this code still looks shared with the Java code, especially if assessCG could be made an abstract method on the FingerprintExtractor trait.